scapy/contrib/tacacs.py

```diff
@@ -0,0 +1,465 @@
+                   ByteEnumField('authen_type', 1, TACACSAUTHENTYPE),
+                   ByteEnumField('authen_service', 1, TACACSAUTHENSERVICE),
+                   FieldLenField('user_len', None, fmt='!B', length_of='user'),
+                   FieldLenField('port_len', None, fmt='!B', length_of='port'),
+                   FieldLenField('rem_addr_len', None, fmt='!B', length_of='rem_addr'),
+                   FieldLenField('arg_cnt', None, fmt='!B', count_of='arg_len_list'),
+                   FieldListField('arg_len_list', [], ByteField('', 0),
+                                  length_from=lambda pkt: pkt.arg_cnt),
+                   StrLenField('user', '', length_from=lambda x: x.user_len),
+                   StrLenField('port', '', length_from=lambda x: x.port_len),
+                   StrLenField('rem_addr', '', length_from=lambda x: x.rem_addr_len)]
+
+    def guess_payload_class(self, pay):
+        if self.arg_cnt > 0:
+            return TacacsPacketArguments
+        return conf.padding_layer
+
+class TacacsAccountingReply(Packet):
+
+    '''
+
+    Tacacs accounting reply body from section 6.2
+    https://tools.ietf.org/html/draft-ietf-opsawg-tacacs-06#section-6.2
+
+    '''
+
+    name = 'Tacacs Accounting Reply Body'
+    fields_desc = [FieldLenField('server_msg_len', None, fmt='!H', length_of='server_msg'),
+                   FieldLenField('data_len', None, fmt='!H', length_of='data'),
+                   ByteEnumField('status', None, TACACSACNTSTATUS),
+                   StrLenField('server_msg', '', length_from=lambda x: x.server_msg_len),
+                   StrLenField('data', '', length_from=lambda x: x.data_len)]
+
+class TacacsPacketArguments(Packet):
+
+    '''
+
+    Class defined to handle the arguments listed at the end of tacacs+
+    Authorization and Accounting packets.
+
+    '''
+
+    __slots__ = ['_len']
+    name = 'Arguments in Tacacs+ packet'
+    fields_desc = [StrLenField('data', '', length_from=lambda pkt: pkt._len)]
+
+    def pre_dissect(self, s):
+        cur = self.underlayer
+        i = 0
+
+        # Searching the position in layer in order to get its length 
+
+        while isinstance(cur, TacacsPacketArguments):
+            cur = cur.underlayer
+            i += 1
+        self._len = cur.arg_len_list[i]
+        return s
+
+    def guess_payload_class(self, pay):
+        cur = self.underlayer
+        i = 0
+
+        # Guessing if Argument packet. Nothing in encapsulated via tacacs+
+
+        while isinstance(cur, TacacsPacketArguments):
+            cur = cur.underlayer
+            i += 1
+        if i+1 < cur.arg_cnt:
+            return TacacsPacketArguments
+        return conf.padding_layer
+
+
+
+class TacacsClientPacket(Packet):
+
+    ''' 
+
+    Super class for tacacs packet in order to get them uncrypted 
+    Obfuscation methodology from section 3.7
+    https://tools.ietf.org/html/draft-ietf-opsawg-tacacs-06#section-3.7
+
+    '''
+
+    def post_dissect(self, pay):
```

Comment:
Can you merge the main logic from `TacacsPacketArguments.post_dissect()` and `TacacsHeader.post_build()` in a function ?